SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1684,35 +1684,72 @@ USpatialActorChannel* ActorSystem::TryRestoreActorChannelForStablyNamedActor(AAc
 
 void ActorSystem::InvokeRepNotifies()
 {
-	for (FObjectRepNotifies& ObjectRepNotifies : RepNotifiesToSend)
+	// We want to call rep notifies on actors before subobjects
+	TArray<FObjectRepNotifies> Subobjects;
+	Subobjects.Reserve(RepNotifiesToSend.Num());
+
+	for (auto& ObjectToObjRepNotifies : RepNotifiesToSend)
 	{
-		UObject* Object = ObjectRepNotifies.Object.Get();
-		if (!IsValid(Object))
+		FObjectRepNotifies& ObjectRepNotifies = ObjectToObjRepNotifies.Value;
+		if (ObjectRepNotifies.RepNotifies.Num() == 0)
 		{
 			continue;
 		}
-		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
-		if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+
+		UObject* Object = ObjectToObjRepNotifies.Key.Get();
+		ObjectRepNotifies.Object = Object;
+
+		if (Cast<AActor>(Object))
 		{
-			// TODO UNR-5785 - Once fixed, change log level to warning
-			UE_LOG(LogActorSystem, Log, TEXT("Failed to invoke rep notifies for an object as its entity id was invalid. Object: %s"),
-				   *Object->GetName());
-			continue;
+			TryInvokeRepNotifiesForObject(ObjectRepNotifies);
 		}
-		USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
-		if (!IsValid(Channel))
+		else
 		{
-			UE_LOG(LogActorSystem, Warning,
-				   TEXT("Failed to invoke rep notifies for an object as its channel was invalid. Object: %s, Entity: %lld"),
-				   *Object->GetName(), EntityId);
-			continue;
+			Subobjects.Add(MoveTemp(ObjectRepNotifies));
```

Comment:
hm, you could also just scan the array twice, calling repnotifies for actors and then for non-actors, which I think would be faster than the allocation?
Another alternative would be to have `RepNotifiesToSend` split into `ActorRepNotifiesToSend` and `SubobjectRepNotifiesToSend` from the start and things would add themselves to the correct one.

Reply:
yea, what would your preference be out of the options

Reply:
Scan array twice > Split notifies > current solution (for readability).
Split notifies > Scan array twice > current solution (for speed).

Reply:
now have two maps SubobjectsRepnotifiestosend and actorrepnotifiestosend